### DISCUSSION TITLE
Runtime  Injection

### DISCUSSION THREAD

**Discussion by din5007** [ctrlplusb/easy-peasy#825] [Q&A]
How can I update my axios interceptor directly with state value ? 
Or how could inject runtime axios instance ?

**Comment by jmyrland** [comment]
If you want to, you can override [injections on the store provider at runtime](https://easy-peasy.dev/docs/api/create-context-store.html#defining-injections-at-runtime).

Alternatively, you could depend on & access the store state within your interceptor

```js
import store from './my-store'; 

const instance = axios.create();

instance.interceptors.request.use(function () {
  store.getState()
});
```
